text/0019-beforeSendTransaction.md

```diff
@@ -0,0 +1,102 @@
+
+# Supporting Data
+
+Issues where this has come up:
+- https://github.com/getsentry/sentry-docs/issues/5525
+- https://github.com/getsentry/sentry-javascript/issues/4723
+- https://github.com/getsentry/sentry-javascript/issues/5442
+- https://github.com/getsentry/sentry-python/issues/1226
+
+(and I'm sure many others)
+
+# Options Considered
+
+_(Revised after preliminary discussion in https://github.com/getsentry/rfcs/pull/19)_
+
+### **Option 1**
+
+_(Original proposal, demoted to one option of several after discussion)_
+
+Add `beforeSendTransaction` to all SDKs, which (as the name implies) would work exactly the same way `beforeSend` does, but would act upon transactions.
+
+Pros:
+- Not a breaking change, so less pain for customers in the short run.
+- Not a breaking change, so no need to wait for a major release to implement it.
+
+Cons:
+- Increased API surface.
+- Code that acts on shared event properties (tags, contexts, etc) would need to be duplicated between `beforeSend` and `beforeSendTransaction`.
+- Unclear/non-parallel naming when considered alongside `beforeSend`. (`beforeSendTransaction` is a clear name, but `beforeSend` is not - it really ought to be `beforeSendError`; similar problem to current confusion with `sampleRate` and `tracesSampleRate`.)
+- Future payload types (replays, profiles, etc) would need their own methods. Where does it end?
+
+### **Option 2**
+
+In the next major of each SDK, start sending both errors and transactions through `beforeSend`.
+
+Pros:
+- `beforeSend` has a seemingly-universally-applicable name, so the most intuitive thing for users is to have it fulfill its implied role and apply universally.
+- Future payload types could  be added without introducing new init options/versions of the method.
+
+Cons:
+- Potentially breaking change for everyone, even people not using tracing, because old assumptions about `beforeSend` input schema would no longer hold. (We could maybe work around this by just surrounding our `beforeSend` call with a `try-catch` and returning the event untouched if things go sideways.)
+
+### **Option 3**
+
+Like option 2, except `beforeSend` doesn't change and instead we create a new universal method to take its place.
+
+Pros:
+- Not a breaking change, so less pain for customers in the short run.
+- Not a breaking change, so no need to wait for a major release to implement it.
+
+Cons:
+- What would we call it? The most logical name given other option names (`beforeBreadcrumb`, `beforeNavigate`, etc) is already taken.
+- What would happen to `beforeSend`? Either we'd eventually either: 1) deprecate and remove it, which just kicks the breaking-change can down the road, or 2) keep it around in perpetutity, which isn't great either, because it would probably continue to confuse people (both by providing a less-functional way to do what the new method does and by _still_ not being named `beforeSendError`).
```

Comment:
I think these are not cons; these are problems to solve. I think having one method called before sending anything makes sense. Alongside adding the universal method, we could also have fine-grained methods. Maybe that would be Option 5, then.

**Option 5**
Add a new universal `beforeSend` method, deprecate `beforeSend`, and add fine-grained `beforeSend` for events and transactions.
* `beforeSendAll` (name can be discussed)
* `beforeSendEvent`
* `beforeSendTransaction`

**Option 6**
Make `beforeSend` the new universal `beforeSend` method, deprecate, and add fine-grained `beforeSend` for events and transactions.
* `beforeSend`
* `beforeSendEvent`
* `beforeSendTransaction`